node/task.ts

```diff
@@ -903,49 +942,25 @@ export function find(findPath: string, options?: FindOptions): string[] {
         while (stack.length) {
             // pop the next item and push to the result array
             let item = stack.pop()!; // non-null because `stack.length` was truthy
-            result.push(item.path);
 
-            // stat the item.  the stat info is used further below to determine whether to traverse deeper
-            //
-            // stat returns info about the target of a symlink (or symlink chain),
-            // lstat returns info about a symlink itself
             let stats: fs.Stats;
-            if (options.followSymbolicLinks) {
-                try {
-                    // use stat (following all symlinks)
-                    stats = fs.statSync(item.path);
-                }
-                catch (err) {
-                    if (err.code == 'ENOENT' && options.allowBrokenSymbolicLinks) {
-                        // fallback to lstat (broken symlinks allowed)
-                        stats = fs.lstatSync(item.path);
-                        debug(`  ${item.path} (broken symlink)`);
-                    }
-                    else {
-                        throw err;
-                    }
-                }
-            }
-            else if (options.followSpecifiedSymbolicLink && result.length == 1) {
-                try {
-                    // use stat (following symlinks for the specified path and this is the specified path)
-                    stats = fs.statSync(item.path);
-                }
-                catch (err) {
-                    if (err.code == 'ENOENT' && options.allowBrokenSymbolicLinks) {
-                        // fallback to lstat (broken symlinks allowed)
-                        stats = fs.lstatSync(item.path);
-                        debug(`  ${item.path} (broken symlink)`);
-                    }
-                    else {
-                        throw err;
-                    }
+            try {
+                // `item.path` equals `findPath` for the first item to be processed, when the `result` array is empty
+                const isPathToSearch = !result.length;
+
+                // following all symlinks OR following symlinks for the specified path AND this is the specified path
+                const followSymbolicLinks = options.followSymbolicLinks || options.followSpecifiedSymbolicLink && isPathToSearch;
+
+                // stat the item. The stat info is used further below to determine whether to traverse deeper
+                stats = _getStats(item.path, followSymbolicLinks, options.allowBrokenSymbolicLinks);
+            } catch (err) {
+                if (err.code == 'ENOENT' && options.skipMissingFiles) {
+                    debug(`File "${item.path}" seems to be removed during find operation execution - so skipping it.`);
```

Comment:
We have to fix this message since the error here can be not only due to the missing file but due to a broken symlink also